src/main/java/org/gridsuite/notification/server/NotificationWebSocketHandler.java

```diff
@@ -110,8 +115,15 @@ public NotificationWebSocketHandler(ObjectMapper jacksonObjectMapper, @Value("${
     @Override
     public Mono<Void> handle(WebSocketSession webSocketSession) {
         URI uri = webSocketSession.getHandshakeInfo().getUri();
-        MultiValueMap<String, String> parameters = UriComponentsBuilder.fromUri(uri).build().getQueryParams();
+        MultiValueMap<String, String> parameters = UriComponentsBuilder.fromUri(uri).build(true).getQueryParams();
         String filterStudyName = parameters.getFirst(QUERY_STUDY_NAME);
+        if (filterStudyName != null) {
+            try {
+                filterStudyName = java.net.URLDecoder.decode(filterStudyName, StandardCharsets.UTF_8.name());
+            } catch (UnsupportedEncodingException e) {
+                LOGGER.error(e.toString(), e);
```

Comment:
let the exception propagates?

Reply:
The method override a method that doesn't throw exception (Mono<Void> handle(WebSocketSession var1);)
I propagated the exception that way } catch (UnsupportedEncodingException e) {
                throw new RuntimeException(e.getCause());
            }
But sonar smells it as a bug...

Reply:
Sonar tells you not to wrap with a runtimeexception not to avoid propagating the exception

Reply:
See all more runtime exception subclasses in powsybl commons

Reply:
I created my exception. Does it worth it to pull a powsybl depedency just for one exception ?